azure-migrate/dependencies-at-scale/README.md

```diff
@@ -0,0 +1,130 @@
+
+## Current Limitations:
+- Each appliance supports enabling Agentless dependency analysis on up to 3000 for VMware type and 1000 for other types.
+
+## Pre-requisites
+Before you get started, you need to do the following:
+- [Review](https://docs.microsoft.com/azure/migrate/migrate-support-matrix-vmware#dependency-analysis-requirements-agentless) supported operating systems, and required permissions for agentless dependency analysis.
+- Have an Azure Migrate project and the Azure Migrate:Server Assessment tool is added to the project.
+- Set up an [Azure Migrate appliance](https://docs.microsoft.com/azure/migrate/migrate-appliance) to discover the on-premises VMware VMs. The appliance discovers on-premises machines, and sends metadata and performance data to Azure Migrate:Server Assessment.
+- Ensure that VMware Tools (later than 10.2) is installed on each VM you want to analyze.
+- Ensure you have downloaded the AzMig_Dependencies powershell module in this folder.
+- Ensure you have PowerShell available. It is recommended you use the [latest version of PowerShell](https://docs.microsoft.com/powershell/scripting/install/installing-powershell).
+- Ensure you have the [Azure PowerShell](https://docs.microsoft.com/powershell/azure/install-az-ps) module installed.
+- The Az.Accounts module is required and must be at least version 2.2.0 or higher.
+- The Az.ResourceGraph module is required and PowerShellGet version 2.0.1 or higher is required.
+
+## Log in to Azure
+
+1. Log into your Azure subscription using the Connect-AzAccount cmdlet.
+
+```PowerShell
+Connect-AzAccount
+```
+
+If using Azure Government, use the following command.
+
+```PowerShell
+Connect-AzAccount -EnvironmentName AzureUSGovernment
+```
+
+2. Select the subscription in which you have created the Azure Migrate project 
+
+```PowerShell
+Select-AZSubscription -subscription "Fabrikam Demo Subscription"
+```
+
+3. Install the Resource Graph module.
+
+```PowerShell
+Install-Module -Name Az.ResourceGraph -Repository PSGallery -Scope AllUsers
+```
+
+4. Import the downloaded AzMig_Dependencies powershell module
+
+```PowerShell
+Import-Module .\AzMig_Dependencies.psm1
+```
+## Enable or disable dependency data collection
+
+1. Get the list of discovered VMware VMs in your Azure Migrate project using the following commands. In the example below, the project name is FabrikamDemoProject, and the resource group it belongs to is FabrikamDemoRG. The list of machines will be saved in FabrikamDemo_VMs.csv
+
+```PowerShell
+Get-AzMigDiscoveredVMwareVMs -ResourceGroupName "FabrikamDemoRG" -ProjectName "FabrikamDemoProject" -OutputCsvFile "FabrikamDemo_VMs.csv" 
+```
+> Please note the project name in the above query is the same one that is displayed in the Azure Portal. 
+
+To filter machines based on specific properties, you can use the following command:
+
```

Comment:
Add one more command here showing that the user can also use appliance name as an optional parameter
Get-AzMigDiscoveredVMwareVMs -ResourceGroupName "FabrikamDemoRG" -ProjectName "FabrikamDemoProject" -OutputCsvFile "FabrikamDemo_VMs.csv" -ApplianceName "FabrikamDemoAppliance"